### PR TITLE
Adopt sbt-circe-org

### PR DESCRIPTION
**Pull request by fthomas**
This is similar to what @armanbilge did in https://github.com/circe/circe-fs2/pull/394 so that releases can be done via CI.

Note that circe-config's version tags currently don't start with a `v` but sbt-typelevel mandates that. So after this is merged, new tags with the `v` prefix that point to the old tags should be added and pushed:
```shell
git tag v0.8.0 0.8.0
git tag v0.7.0 0.7.0
...
```
Without that, `version` and `mimaPreviousArtifacts` for example won't be set correctly.

### CODE REVIEW

**Comment by armanbilge** [comment]
ScalaCheck had the same issue with old tags, here's a workaround for MiMa.
https://github.com/typelevel/scalacheck/blob/fc50bed0f75a7cc49ce382ed20a6ecaad644bb2a/build.sbt#L56-L67

**Review by armanbilge — APPROVED** [review]
Besides the mima version thing mentioned above, LGTM AFAICT 👍 thank you so much!

**Review by zmccoy — DISMISSED** [review]
Other than the mima fix that @armanbilge mentioned this looks great!  Thank you so much

File: build.sbt
Comment (zmccoy) [inline build.sbt:3]:
```suggestion
ThisBuild / organizationName := "Circe"
```

This would probably be good to change to Circe.

File: config/src/main/scala-2.12/main/scala/io.circe.config/compat/package.scala
Comment (zmccoy) [inline config/src/main/scala-2.12/main/scala/io.circe.config/compat/package.scala:1]: I think we have an extra main/scala in the dir structure of this and the scala-2.13 dir.

File: build.sbt
Comment (fthomas) [reply build.sbt:3]: I think sbt-circe-org just sets this to `circe` but that overwrites the existing license headers on `sbt headerCreate`. That's the reason why I overwrote it here.

Comment (armanbilge) [reply build.sbt:3]: If we're not ok to change the copyright headers, there's probably a workaround.

Comment (armanbilge) [reply build.sbt:3]: Right, you can override the `headerLicense` setting.
https://github.com/typelevel/cats-effect/blob/5f7b24395d8a1df9ddbb5440025ba85c21ce8854/project/Common.scala#L34-L36

File: config/src/main/scala-2.12/main/scala/io.circe.config/compat/package.scala
Comment (fthomas) [reply config/src/main/scala-2.12/main/scala/io.circe.config/compat/package.scala]: Well spotted! Dropped in https://github.com/circe/circe-config/pull/361/commits/0d5bed576b84c0a866be4d5ca33a0343fe0ad8ff.

File: build.sbt
Comment (fthomas) [reply build.sbt]: I tried `headerLicense` before and scoped it with `ThisBuild` but that had no effect. Putting the setting in the project worked. Thanks!

Comment (zarthross) [reply build.sbt]: @zmccoy and I discussed, and we think the right thing to do is to set the `organizationName` to `Circe` and NOT override the existing headers, by adding double attribution headers.   You can do this by adding a blank block comment to each of the files with existing headers and re-running the header creation.  See this[ http4s file](https://github.com/http4s/http4s/blob/0e253e30bf983545cb0ebdd6407495b629f4ed1e/core/shared/src/main/scala/org/http4s/ResponseCookie.scala#L22) for an example.

In addition to the double headers, we think adding a notices file like [this one ](https://github.com/http4s/http4s/blob/0e253e30bf983545cb0ebdd6407495b629f4ed1e/NOTICE) that references the original licensed work [here](https://github.com/jonas/circe-config) would be best.

File: build.sbt
Comment (zarthross) [reply build.sbt]: I agree with @armanbilge, since this is still 'early-semvar' maintaining compat with the last minor version is sufficient.

Comment (fthomas) [reply build.sbt]: CI says it's fine. :-) I've also set `tlVersionIntroduced` which should trim that `Set` appropriately, right? I guess because of `tlVersionIntroduced` we could remove versions less than `0.3.0` here. The current `master` also checks bincompat since `0.3.0`.

Comment (armanbilge) [reply build.sbt]: > I've also set `tlVersionIntroduced` which should trim that `Set` appropriately, right?

I can't remember tbh. I think that trims the information derived from tags, but anything added to `tlMimaPreviousVersions` has final say.

The definitive thing is to check `mimaPreviousArtifacts` and see what's in there.

Anyway, so long as CI is happy there's no problem with guaranteeing _extra_ compatibility beyond semver 😁  Cats does this to guarantee bincompat with 1.x for various modules, even in the 2.x series.

File: build.sbt
Comment (fthomas) [reply build.sbt]: That's a good suggestion. Done in 15a73d1 and 993f07a.

**Review by zmccoy — DISMISSED** [review]
Thanks again!

**Comment by fthomas** [comment]
Sorry @zmccoy for dismissing your last review, but I noticed that my changes removed coverage reporting from CI.

I thought that my last commit would bring it back, but the upload of the coverage report failed with:
```
[2023-02-06T19:36:38.212Z] ['error'] There was an error running the uploader: Flags must consist only of alphanumeric characters, '_', '-', or '.' and not exceed 45 characters. Received 2.12.15,temurin@11
```

**Comment by fthomas** [comment]
Coverage reporting should be fixed with the next sbt-circe-org version but I don't think this is a blocker for this PR.

**Review by zarthross — APPROVED** [review]

**Pull request merged**
